### ISSUE TITLE
Error when compiling due to redefinition of int

### ISSUE THREAD

**Issue by hmtheboy154** [linux-surface/intel-precise-touch#22]
https://github.com/linux-surface/intel-precise-touch/blob/8abe26863ae77aab278d070a8a4604f00b6ec698/src/resources.c#L55

When compiling with AOSP clang, it will show redefinition error, so I have to fix it like this
https://github.com/hmtheboy154/Darkmatter-kernel/commit/8f13146d61ed12676ef2ee094641f73e21c3eb36

**Comment by StollD** [comment]
The reason why this worked before was that the linux kernel didn't allow declaring a variable inside of the for statement. This was a somewhat recent change (less than a year ago I think) that I adopted because it makes the code a bit nicer.

I don't ususally test older kernels, so thank you for bringing this to my attention! I applied your patch, with some slight modifications so that I don't forget why that part of the code is different.

Although it is probably a good idea for me to move the other loop counters out of the for statement again too. Because if your compiler / kernel is still building for an older C standard, this might actually be undefined behaviour.
